signalrkore/src/commonMain/kotlin/eu/lepicekmichal/signalrkore/HubCommunication.kt

```diff
@@ -15646,27 +15652,29 @@ abstract class HubCommunication {
         return receivedInvocations
             .run {
                 if (!hasResult) {
-                    this.onEach {
-                        if (it is HubMessage.Invocation.Blocking) {
-                            logger.log(
-                                level = Logger.Level.WARNING,
-                                message = "There is no result provider for ${it.target} despite server expecting it.",
-                            )
-                            complete(
-                                HubMessage.Completion.Error(
-                                    invocationId = it.invocationId,
-                                    error = "Client did not provide a result."
-                                ),
-                            )
+                    this
+                        .filter { it.target == target }
```

Comment:
Here we have to apply the target filter before the result check, otherwise will be executed for all targets